app/models/authentication.rb

```diff
@@ -158,20 +158,14 @@ def set_tenant_from_group
     self.tenant_id = miq_group.tenant_id if miq_group
   end
 
-  def set_credentials_changed_on
-    return unless @auth_changed
-    self.credentials_changed_on = Time.now.utc
-  end
-
   def after_authentication_changed
-    return unless @auth_changed
     _log.info("[#{resource_type}] [#{resource_id}], previously valid on: [#{last_valid_on}]")
 
     raise_event(:changed)
+    self.credentials_changed_on = Time.now.utc
```

Comment:
Doesn't this need to be done before save?

Reply:
@bdunne It's not clear to me what the semantics are supposed to be. Were the credentials "changed" on a create?

Reply:
It was set in the `before_save`.

Reply:
Right, but logically it seemed like it was only meant to trigger on an update. That's part of the overall issue really.

Reply:
but now it won't be persisted, correct?

Reply:
@bdunne Not sure what you mean by persisted here. Can you show me an example on the rails console or something that shows the behavior you're expecting vs what this PR does? Or a spec perhaps?

Reply:
If this is done in an `after_update` without calling `save` or `update`, it won't write the data into the database, correct?